src/server.rs

```diff
@@ -297,10 +302,81 @@ async fn upload(
     Ok(HttpResponse::Ok().body(urls.join("")))
 }
 
+#[derive(Serialize, Deserialize)]
+struct ListItem {
+    file_name: String,
+    file_size: u64,
+    expires_at: Option<String>,
+}
+
+// Returns a list of files
+#[get("/list")]
+async fn list(
+    request: HttpRequest,
+    config: web::Data<RwLock<Config>>,
+) -> Result<HttpResponse, Error> {
+    let config = config
+        .read()
+        .map_err(|_| error::ErrorInternalServerError("cannot acquire config"))?
+        .clone();
+
+    if !config.server.json_list_enabled.unwrap_or(false) {
+        Err(error::ErrorForbidden("json list is not enabled"))?;
+    }
+
+    let connection = request.connection_info().clone();
+    let host = connection.realip_remote_addr().unwrap_or("unknown host");
+
+    auth::check(
+        host,
+        request.headers(),
+        env::var(AUTH_TOKEN_ENV)
+            .ok()
+            .or_else(|| config.server.auth_token.as_ref().cloned()),
+    )?;
```

Comment:
Here is the change for the auth call.

```suggestion
     let tokens = config.get_tokens();
     auth::check(host, request.headers(), tokens)?;
```